Library/ENV/4.3/cc

```diff
@@ -194,6 +196,27 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar)
+      %r[#{cellar}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(opt)
+      %r[#{opt}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(prefix)
+      true
+    else
+      !path.start_with?("/opt", "/sw", "/usr/X11")
```

Comment:
Actually, I think this is going to conflict with at least a few things - things would need to get sorted to make this change. e.g. here in sdl: https://github.com/Homebrew/homebrew/blob/369bcfc6179ca4393fa430eccdee58e75f90e6fe/Library/Formula/sdl.rb#L59-L62